parlai/core/torch_generator_agent.py

```diff
@@ -1611,7 +1714,22 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # get the actual word id from residual of the same division
         tok_ids = best_idxs % voc_size
 
-        return (hyp_ids, tok_ids, best_scores)
+        tok_scores = torch.gather(logprobs, 1, tok_ids.unsqueeze(0)).view(-1)
+
+        tok_ranks = (
```

Comment:
I think this should work. We compute ranks along each beam by using double argsort along beams (dimension 1). Then, we flatten to align the output of this with `flat_beam_scores`. Finally, we use `best_idxs` to extract the relevant ranks.